bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -460,6 +553,176 @@ impl FromHexStr for Height {
     }
 }
 
+impl ops::Add<relative::Height> for Height {
+    type Output = Height;
+    #[inline]
+    fn add(self, rhs: relative::Height) -> Self::Output {
+        match self.checked_add(rhs.value().into()) {
+            Some(height) => height,
+            None => panic!("attempted to add height with overflow"),
+        }
+    }
+}
+
+impl ops::Add<u32> for Height {
+    type Output = Height;
+    #[inline]
+    fn add(self, rhs: u32) -> Self::Output {
+        let height = self.0 + rhs;
+        Height::from_consensus(height).expect("attempted to add with overflow")
+    }
+}
+
+impl ops::Add<Height> for u32 {
+    type Output = Height;
+    #[inline]
+    fn add(self, rhs: Height) -> Self::Output { rhs + self }
+}
+
+impl ops::Sub<relative::Height> for Height {
+    type Output = Height;
+    #[inline]
+    fn sub(self, rhs: relative::Height) -> Self::Output {
+        match self.checked_sub(rhs.value().into()) {
+            Some(height) => height,
+            None => panic!("attempted to sub height with overflow"),
+        }
+    }
+}
+
+impl ops::Sub<u32> for Height {
+    type Output = Height;
+    #[inline]
+    fn sub(self, rhs: u32) -> Self::Output {
+        let height = self.0 - rhs;
+        Height::from_consensus(height).expect("non-panicing sub as a valid height")
+    }
+}
+
+impl ops::Sub<Height> for u32 {
+    type Output = Height;
+    #[inline]
+    fn sub(self, rhs: Height) -> Self::Output { rhs - self }
+}
+
+impl ops::Mul<relative::Height> for Height {
+    type Output = Height;
+    #[inline]
+    fn mul(self, rhs: relative::Height) -> Self::Output {
+        match self.checked_mul(rhs.value().into()) {
+            Some(height) => height,
+            None => panic!("attempted to mul height with overflow"),
+        }
+    }
+}
+
+impl ops::Mul<u32> for Height {
```

Comment:
Note that while I was in favor of this in other places I don't think it's appropriate here. Multiplying by a scalar is scaling but multiplying an instant doesn't seem to make sense. Think of height like specific date and time. What does `(January 3rd 2009) * 42` mean?